repl/main.js

```diff
@@ -6,17 +6,47 @@ const outputTemplate = document.getElementById("repl-output-template");
 const inputElement = document.getElementById("input");
 const inputTextArea = inputElement.querySelector("textarea");
 const outputElement = document.getElementById("repl-contents");
+const loadingContainer = document.getElementById("loading-content");
+const mainContainer = document.getElementById("main-content");
+const loadingText = document.getElementById("loading-text");
+const loadingProgress = document.getElementById("loading-progress");
+const headerDescriptionSpan = document.getElementById("header-description");
 
 (async function () {
+  function updateLoading(name, { loaded, total, known }) {
+    loadingText.innerText = `Loading ${name}...`;
+    if (known) {
+      loadingProgress.max = total;
+      loadingProgress.value = loaded;
+    } else {
+      delete loadingProgress.max;
+      delete loadingProgress.value;
+    }
+  }
+
+  updateLoading("Build Information", { known: false });
+  fetch("/repl/commit")
```

Comment:
I think ideally we'd integrate this in libjs.js - I assume you're not manually copying a git commit into the file every time, so perhaps we could simply append something like `Module.SERENITYOS_COMMIT = "xxx";` to the file when it's being generated?

Reply:
I could, the build job just echoes it to that `commit` file atm

Reply:
over here: [SerenityOS/serenity@`dac1685` (#16133)](https://github.com/SerenityOS/serenity/pull/16133/commits/dac1685e5c3517fcd0c9924b97026db6c6f1789e#diff-7985cab97a05bfe5fc9e7b10f0a11e6889e2eddf3f937bb16f85e14409e3b35eR89)

Reply:
I'll just make it append the commit hash into `libjs.js`, that seems like the best option.